en/material-system/effect-2.x-to-3.0.md

```diff
@@ -0,0 +1,254 @@
+
+- Technique list of rendering techniques
+- A list of passes for each Technique
+- A list of editable properties exposed to the editor in each Pass (including data type designation within the editor, component mapping relationships, etc.)
+- Shader programs for each Pass, including vertex and slice shader programs
+
+In terms of syntax details, such as Property declarations and macro definitions, the approach is the same:
+
+![effect](material4.jpg)
+
+![effect](material4.1.jpg)
+
+![effect](material5.jpg)
+
+### 2.2 Preset Material Types
+
+There is a big difference between v2.x and v3.0 in terms of preset materials.
+
+- The preset materials in v2.x include 2D Sprite, classic `blinn-phong` lighting materials, `unlit` materials, default `toon` materials, particle materials, etc.
+
+- The preset materials in v3.0 are physics-based rendering system, including standard PBR materials based on physical lighting, Skybox, toon rendering materials, 3D particle materials (CPU & GPU), particle trailing materials, traditional 2D Sprite materials, etc.
+
+The v3.0 default standard material supports the standard Physically Based Rendering (PBR) process, which contains a lot of mapping information to enhance the quality and realism of the material, such as diffuse mapping, normal mapping, metallic, roughness, ambient light oclusion, and so on. The whole algorithm is based on the standard BRDF lighting model, which is not available in v2.x. The overall performance of v3.0 is a step up from v2.x.
+
+### 2.3 Effect Writing Details Differences
+
+Although the syntax rules of Effect in Cocos Creator are basically the same in v2.x and v3.0, there are still differences in many built-in header files, variable names and function names.
+
+For example, in v3.0, it was `cc_mainLitDir` and the header file `cc-global` was included. In v2.x, you need to use `cc_lightDirection[i]` to get the direction of the light source, and you need to include the header `cc-lights`. See the third point below in the **API Upgrade Guide** section for more details on the differences.
+
+Some default shader functions are unique to v3.0, such as `CCStandardShading`, `CCToonShading`, etc., which are not available on v2.x. Again, see the third point below in the **API Upgrade Guide** section.
+
+Regarding uniform declarations, v3.0 forces the use of UBO for organization, and the minimum unit for memory layout is vec4, which no longer supports separate declarations of float or vec3 types of uniform.
+
+In terms of header files, v3.0 has built-in editor header assets in the `assets/chunks` directory of **Internal DB**. They can be referenced directly without adding directories, mainly including some common **tool functions** and **standard lighting model**. The header files of v2.x are built into the editor, so you can't see what they are.
```

Comment:
```suggestion
In terms of header files, v3.0 has built-in editor header assets in the `assets/chunks` directory of **Internal DB**. They can be referenced directly with file name instead of file path, mainly for including some common **tool functions** and **standard shading functions**. The header files of v2.x are built into the editor, so you can't see what they are.
```